Read/Edit/Write timeline files

Ronserruya opened the pull request:
Added the ability to read/modify/write timeline files

The entry point to the whole thing is the `CharaTimelineFile` file, but most of the time you'd interact with the `CharaTimelineFile.Timeline` object

The "framework" allows you to read and modify any timeline file, but you can only write it back if all elements of the file are recognized (meaning we know their exact internal structure)

Adding new ElementTypes is pretty simple, just need to create a new class for the element with all the fields, all the read/write logic is done in the base class "BaseStruct" with reflection.

The only exceptions (so far) are classes that contain a list of structs or internal sub-structs as fields (8, 84, 1023, 1030, 1049). I could handle these in a generic way as well, but the code already became kind of a mess, so I just did it separately for these


Writing with all the offsets was honestly more complex than I thought it would be, so I wonder if I just overcomplicated it 😅 

I probably over-abused Linq, but I was kind of learning C# while writing all of this, and I found this feature really cool, and it can read all the timeline files in the game in like a minute, so optimizing isn't needed 😅 

The next step with this is to figure out more element types, since most files contain some types we don't know, then make editing easier, like with dump/load to json, or some GUI editor

![Screenshot 2025-05-07 035507](https://github.com/user-attachments/assets/7b0ce3ba-348b-426e-8975-d8b719afd1c6)

Nenkai commented:
Thanks for this!

I would say what isn't ideal is:
* The many intermediate structures used for representing each actual game structure
* Linq usage certainly did overcomplicate imo - but for now it's fine
* Generally you want a class per file. Which means one file per element type.

But for now it's fine. Will point out other things in other comments

Ronserruya commented:
@Nenkai Great, i'll check it out once you are finished
FYI I fixed some issues now and added two new Elements (29 and 1008)
I don't know their internal structure but I managed to read/write them regardless

Ronserruya commented:
I added most of the changes, but left the Elements in the same file as they are for now.
I also created a new FF16Tools.Files.Timelines namespace, where I put the Element types in  

And moved all the rest of the code under FF16Tools.Files.Timelines.Chara
